misc/keyvaluestorage/src/node-js/index.ts

```diff
@@ -1,62 +1,62 @@
-import Database, { Database as IDatabase } from "better-sqlite3";
+
 import { safeJsonParse, safeJsonStringify } from "safe-json-utils";
 
 import {
   IKeyValueStorage,
   KeyValueStorageOptions,
-  parseEntry,
 } from "../shared";
-import { Statements } from "./sqlite";
+import db from "./level_db"
 
-const DB_NAME = "walletconnect_kvs.db";
-const TABLE_NAME = "keyvaluestorage";
+const DB_NAME = "walletconnect_level.db";
 
 export class KeyValueStorage implements IKeyValueStorage {
-  private readonly database: IDatabase;
-  private readonly statements: Statements;
+  private database;
 
   constructor(opts?: KeyValueStorageOptions) {
-    this.database = new Database(opts?.database || DB_NAME);
-    this.statements = new Statements(opts?.table || TABLE_NAME);
-    this.database.prepare(this.statements.createTable()).run();
+    this.database = db.create(opts?.database || opts?.table || `${DB_NAME}`)
+    if(!opts || opts.database == ':memory:') {
+      this.database.clear()
+    }
   }
 
   public async getKeys(): Promise<string[]> {
-    const keys = this.database
-      .prepare(this.statements.selectKeys())
-      .all()
-      .map<string[]>(x => Object.values(x))
-      .flat();
-    return keys;
+    const entries = (await this.database.iterator().all()).map(([key]) => key as string);
+    return entries
   }
 
   public async getEntries<T = any>(): Promise<[string, T][]> {
-    const entries = this.database
-      .prepare(this.statements.selectEntries())
-      .all()
-      .map(x => parseEntry(Object.values(x) as [string, any]));
+  
+    const entries = (await this.database.iterator().all()).map(([key, value]) => [key, safeJsonParse(value)] as [string, any]);
```

Comment:
Think the cast at the end needs to be a reference to `T` not `any` to respect type generics:

```suggestion
    const entries = (await this.database.iterator().all()).map(([key, value]) => [key, safeJsonParse(value)] as [string, T]);
```

I see it was any before but worth seeing if this works since it seems to be the correct assertion 🤔